website/pages/component/[id]/index.tsx

```diff
@@ -61,7 +61,9 @@ export const getStaticProps: GetStaticProps = async ({ params }) => {
 
   const { id } = params;
 
-  const props: Partial<BaseLayoutProps> = { component: id };
+  const props: Partial<BaseLayoutProps> = {
+    component: id as BaseLayoutProps['component'],
```

Comment:
if we used `params.id` directly here, would TS know it's a string since we had the `typeof` check above?

Reply:
it knows it's a string but not that it belongs to a specific subset of strings